### PR TITLE
{vis}[foss/2016a] cairo v1.14.6, freetype v2.6.2, fontconfig v2.11.94, ...

### PR DESCRIPTION
**Pull request by SimonPinches**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
gligar01.gligar.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c8b239505a2ba3602419 for a full test report.

File: easybuild/easyconfigs/c/cairo/cairo-1.14.6-foss-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+sources = [SOURCE_TAR_XZ]
+
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('zlib', '1.2.8'),
+    ('libpng', '1.6.21'),
+    ('freetype', '2.6.2'),
+    ('pixman', '0.34.0'),
+    ('fontconfig', '2.11.94'),
+    ('expat', '2.1.0'),
```
Comment (boegel) [inline easybuild/easyconfigs/c/cairo/cairo-1.14.6-foss-2016a.eb]: @SimonPinches our most recent `cairo` easyconfigs also have `GLib` and `libX11` listed as dependencies; did you remove these intentionally here?

cfr. https://github.com/hpcugent/easybuild-easyconfigs/blob/master/easybuild/easyconfigs/c/cairo/cairo-1.14.6-intel-2016a.eb

File: easybuild/easyconfigs/c/cairo/cairo-1.14.6-foss-2016a.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+]
+
+# disable symbol lookup, which requires -lbfd, to avoid link issues with (non-PIC) libiberty.a provided by GCC
+configopts = "--enable-symbol-lookup=no"
```
Comment (boegel) [inline easybuild/easyconfigs/c/cairo/cairo-1.14.6-foss-2016a.eb]: also missing several configure options, a `buildopts` line and a custom `sanity_check_paths`

**Comment by hajgato** [comment]
@SimonPinches @boegel  Check PR #2761

File: easybuild/easyconfigs/c/cairo/cairo-1.14.6-foss-2016a.eb (hunk 1 above)
Comment (SimonPinches) [inline easybuild/easyconfigs/c/cairo/cairo-1.14.6-foss-2016a.eb]: No, I didn't deliberately remove then - I just based the new easyconfig file on 'cairo-1.14.2-goolf-1.7.20.eb' which didn't have them either.  Putting them back in revealed lots of other missing (build) dependencies which I've now also added.  Again, the benefits of trying to build on CentOS 5 ;-)

**Comment by hajgato** [comment]
@SimonPinches I do not understand your effort here. All this easyconfigs are done by me in various PRs, but they not yet merged.

**Comment by hajgato** [comment]
@SimonPinches check PR #2761

**Comment by hajgato** [comment]
@SimonPinches check PR #2758

**Comment by hajgato** [comment]
@SimonPinches check PR #2757

**Comment by SimonPinches** [comment]
Hi @hajgato, I certainly didn't want to duplicate any effort (I'm sure neither of us have the time).  I just added the new easyconfigs I needed for the build to succeed (or try to succeed).  Since your changes aren't (or weren't) in the last release I didn't know they existed.

**Comment by SimonPinches** [comment]
Jenkins seems to report a dependency tangle but I'm not sure exactly what the problem is or how to fix it...

File: easybuild/easyconfigs/l/libXrender/libXrender-0.9.9-foss-2016a.eb [hunk 3]
```diff
@@ -14,6 +14,10 @@ source_urls = [XORG_LIB_SOURCE]
 builddependencies = [
     ('kbproto', '1.0.7'),
     ('renderproto', '0.11'),
+    ('libX11', '1.6.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/l/libXrender/libXrender-0.9.9-foss-2016a.eb]: @SimonPinches the problem indicated by Jenkins is that you're including `libX11` both as build depenency and runtime dependency (it already was a runtime dep), so: just drop this line

**Comment by boegel** [comment]
@hajgato I'll look into merging your PRs first; as @SimonPinches pointed out, he's working on an old OS, which may have pointed out some missing dependencies; those will stand out once your PRs are merged, and this branch is synced with `develop` then (I can help with that if needed @SimonPinches)

I'll get back to this once #2758 and #2761 are merged (#2757 is merged already)

**Comment by boegel** [comment]
@SimonPinches sync with current `develop` available in https://github.com/SimonPinches/easybuild-easyconfigs/pull/2

**Comment by boegel** [comment]
@hajgato thoughts on the additional (build) deps?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2615.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/77aed8a0a7afb45693633075c053fb4f for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2120.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/d0db9a8d60131133e4b29e811ac6f94f for a full test report.

**Comment by hajgato** [comment]
@boegel: I can confirm `xorg-macros` but I think `libpthread-stubs` and `xproto` are not needed. (Although when you get into this point, probably you already have them).
This is why I was working on PR #2579 , but that project a bit dead due to the "this easyblock does too few things and includes too much copy-pasting" error. :D (https://github.com/hpcugent/easybuild-easyblocks/pull/843), beacuse we would install all X stuff in one big package.

**Comment by SimonPinches** [comment]
I've just tested again and I cannot install `libXrender/0.9.9-foss-2016a` if I comment out either of the builddependencies `libpthread-stubs` or `xproto` in `libXrender-0.9.9-foss-2016a.eb` :-(

The two log files are here: [Uploading easybuild-libXrender-0.9.9-20160412.114054.RJtBV.txt…]() and 
[easybuild-libXrender-0.9.9-20160412.113957.fnVZy.txt](https://github.com/hpcugent/easybuild-easyconfigs/files/214725/easybuild-libXrender-0.9.9-20160412.113957.fnVZy.txt)

**Comment by hajgato** [comment]
@SimonPinches @boegel Drat and double drat. It seems that I have those protos installed on the system. (really need to do the "bulk" x installation!)

**Comment by boegel** [comment]
OK, then I guess this is good to go.

This does need another sync with `develop` though, the merge is broken due to a merge conflict... @SimonPinches?

**Comment by boegel** [comment]
@SimonPinches merge conflict fixed in https://github.com/SimonPinches/easybuild-easyconfigs/pull/3

**Comment by boegel** [comment]
lgtm, submitted a final test build

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2725.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/9cee76ec30dcddc3f8f5e027ce8c1eec for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node2128.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e77479ed1d00aa51b033d0135d85d436 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @SimonPinches!

**Pull request merged**